11_bank_card.py

```diff
@@ -0,0 +1,166 @@
+"""
+Hometask 21 Bank card operation.
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters,
+# __str_ magic method
+# do not forget about block
+
+"""
+
+import datetime
+import random
+
+
+class BankCard:
+    """Make class."""
+
+    bank = 'Best Bank'
+    card = 'VISA'
+
+    def __init__(self, name, surname, balance):
+        """Make init."""
+        self.name = name
+        self.surname = surname
+        self._balance = balance
+        self.__card_num = self.get_card_number
+        self._exp_date = self.exp_date()
+        self._card_pin = self.card_pin()
+        self._card_cvv = self.card_cvv()
+
+    @staticmethod
+    def get_card_number():
+        """secret_card = f"{'*' * 12}{card[-4:]}."""
+        """card = str(random.randint(1000000000000000, 9999999999999999))
+        it's also works."""
+        """card = str(random.randint(10 ** 15, (10 ** 16) - 1))"""
+        """f"{'*' * 12}{card[-4:]}" """
+        return str(random.randint(10 ** 15, (10 ** 16) - 1))
+
+    @staticmethod
+    def exp_date():
+        """Find exp date."""
+        today = datetime.date.today()
+        exp_month = today.month
+        exp_year = int(str(today.year + 4)[2:])
+        return f'{exp_month:02d}/{exp_year}'
+
+    @staticmethod
+    def card_pin():
+        """Make static method."""
+        card_pin = random.randint(1000, 9999)
+        return card_pin
+
+    @staticmethod
+    def card_cvv():
+        """Find random cvv code."""
+        return random.randint(100, 999)
+
+    @property
+    def cvv(self):
+        """Getter."""
+        return self._card_cvv
+
+    @cvv.setter
+    def cvv(self, value):
+        """Setter."""
+        self._card_cvv = value
+
+    @property
+    def pin(self):
+        """Getter."""
+        return self._card_pin
+
+    @pin.setter
+    def pin(self, value):
+        """Setter."""
+        self._card_pin = value
+
+    @property
+    def ex_date(self):
+        """Getter."""
+        return self._exp_date
+
+    @ex_date.setter
+    def ex_date(self, value):
+        """Setter."""
+        self._exp_date = value
+
+    @classmethod
+    def get_account_info(cls):
+        """Bank account."""
+        return f'Bank is {cls.bank}'
+
+    @classmethod
+    def get_card_info(cls):
+        """Card type."""
+        return f"Card's type is {cls.card}"
+
+    @classmethod
+    def verification_pin(cls, card_pin, enter_pin):
+        """Check pin code."""
+        if card_pin != enter_pin:
+            raise ValueError('Incorrect PIN code')
+        else:
+            print('Entered PIN code is correct')
+
+    def transfer_money(self, balance, enter_pin, card_pin=1111):
```

Comment:
Here is some hack with card pin. You need to set not random but fixed pin at the beginning and for pin verification jsut check fixed card pin with pin defined by user

Reply:
yes, I thought about it, but for some reason I thought it should have been random...

I corrected all the comments, thank you.